features/developer_portal/login.feature

```diff
@@ -4,12 +4,34 @@ Feature: Login feature
 
   Background:
     Given a provider "foo.3scale.localhost"
-      And provider "foo.3scale.localhost" has multiple applications enabled
-      And a buyer "bob" signed up to provider "foo.3scale.localhost"
+    And provider "foo.3scale.localhost" has multiple applications enabled
+    And a buyer "bob" signed up to provider "foo.3scale.localhost"
+    And the current domain is "foo.3scale.localhost"
 
   @security
   Scenario: Buyer can log in with csrf protection enabled
-    Given the current domain is foo.3scale.localhost
     When I go to the login page
-     And I fill in the "bob" login data
+    And I fill in the "bob" login data
     Then I should be logged in the Development Portal
+
+  @recaptcha
+  Scenario: Captcha is disabled
+    Given provider "foo.3scale.localhost" has "spam protection level" set to "none"
+    When I go to the login page
+    Then I should not see the captcha
+    And I fill in the "bob" login data
+    And I should be logged in the Development Portal
+
+  @recaptcha
+  Scenario: Captcha is forced
+    Given provider "foo.3scale.localhost" has "spam protection level" set to "captcha"
+    When I go to the login page
+    Then I should see the captcha
+
+  @recaptcha
+  Scenario: Spam check fails (because of fast fill and no js)
+    Given provider "foo.3scale.localhost" has "spam protection level" set to "auto"
+    And I go to the login page
+    When timestamp spam check will return probability 1
+    And I fill in the "bob" login data
+    Then I should see the captcha
```

Comment:
Perhaps test login with correct and incorrect captcha data by using steps `I fill in the captcha correctly` and `I fill in the captcha incorrectly`. Other than that PR looks great to me.